src/map.cpp

```diff
@@ -643,16 +643,16 @@ bool Map::isSightClear(const Position& fromPos, const Position& toPos, bool same
 	return checkSightLine(fromPos.x, fromPos.y, toPos.x, toPos.y, fromPos.z);
 }
 
-const Tile* Map::canWalkTo(const Creature& creature, const Position& pos) const
+const std::shared_ptr<Tile> Map::canWalkTo(std::shared_ptr<const Creature> creature, const Position& pos) const
 {
-	Tile* tile = getTile(pos.x, pos.y, pos.z);
-	if (creature.getTile() != tile) {
+	auto tile = getTile(pos.x, pos.y, pos.z);
```

Comment:
Could this be `const auto`
```suggestion
	const auto tile = getTile(pos.x, pos.y, pos.z);
```